grails-app/controllers/org/pih/warehouse/api/StockMovementApiController.groovy

```diff
@@ -496,4 +532,195 @@ class StockMovementApiController {
+            }
+        }
+
+        if (shipmentItems) {
+            def csv = getShipmentItemsCsv(shipmentItems)
+            def date = new Date()
+            response.contentType = "text/csv"
+            response.setHeader("Content-disposition", "attachment; filename=\"Items shipped not received_${destination.name}_${date.format("yyyyMMdd-hhmmss")}.csv\"")
+            render(contentType: "text/csv", text: csv.out.toString())
+            return
+        } else {
+            def message = "No shipment items found"
+            response.status = 404
+            render([errorMessage: message] as JSON)
+            return
+        }
+    }
+
+    /**
+     * Action to get pending requisition items as csv
+     * */
+    def pendingRequisitionItems = {
+        if (!params.origin) {
+            def message = "Origin parameter cannot be the empty"
+            response.status = 400
+            render([errorMessages: [message]] as JSON)
+            return
+        }
+
+        def origin = Location.get(params.origin)
+
+        if (!origin) {
+            def message = "Origin does not exist"
+            response.status = 404
+            render([errorMessage: message] as JSON)
+            return
+        }
+
+        def pendingRequisitionItems = stockMovementService.getPendingRequisitionItems(origin)
+
+        if (pendingRequisitionItems) {
+            def csv = getPendingRequisitionItemsCsv(pendingRequisitionItems)
+            def date = new Date()
+            response.contentType = "text/csv"
+            response.setHeader("Content-disposition", "attachment; filename=\"PendingShipmentItems-${new Date().format("yyyyMMdd-hhmmss")}.csv\"")
+            render(contentType: "text/csv", text: csv.out.toString())
+            return
+        } else {
+            def message = "No pending requisition items found"
+            response.status = 404
+            render([errorMessage: message] as JSON)
+            return
+        }
+    }
+
+    def getShipmentItemsCsv(List shipmentItems) {
+        def csv = CSVUtils.getCSVPrinter()
+        csv.printRecord(
+            "Code",
+            "Product Name",
+            "Quantity Incoming",
+            "Expected Shipping Date",
+            "Expected Delivery Date",
+            "Shipment Number",
+            "Shipment Name",
+            "Origin",
+            "Destination",
+        )
+
+        shipmentItems.each { shipmentItem ->
+            csv.printRecord(
+                shipmentItem.productCode,
+                shipmentItem.productName,
+                shipmentItem.quantity,
+                shipmentItem.expectedShippingDate,
+                shipmentItem.expectedDeliveryDate,
+                shipmentItem.shipmentNumber,
+                shipmentItem.shipmentName,
+                shipmentItem.origin,
+                shipmentItem.destination,
+            )
+        }
+
+        return csv
+    }
+
+    def getPendingRequisitionItemsCsv(List pendingRequisitionItems) {
```

Comment:
Here, these are just requisition items. No need for the pending descriptor. What if I end up decided to pass a list of picked or issued requisition items. It doesn't seem to be doing anything special because of the "pending" status so it's probably not necessary.